pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java

```diff
@@ -377,25 +382,49 @@ public CompletableFuture<Optional<BrokerLookupData>> assign(Optional<ServiceUnit
             if (topic.isPresent() && isInternalTopic(topic.get().toString())) {
                 owner = serviceUnitStateChannel.getChannelOwnerAsync();
             } else {
-                owner = getOwnerAsync(serviceUnit, bundle, false).thenApply(Optional::ofNullable);
+                // Check if this is Heartbeat or SLAMonitor namespace
+                String candidateBroker = NamespaceService.checkHeartbeatNamespace(serviceUnit);
+                if (candidateBroker == null) {
+                    candidateBroker = NamespaceService.checkHeartbeatNamespaceV2(serviceUnit);
+                }
+                if (candidateBroker == null) {
+                    String broker = NamespaceService.getSLAMonitorBrokerName(serviceUnit);
+                    // checking if the broker is up and running
+                    if (broker != null && pulsar.getNamespaceService().isBrokerActive(broker)) {
+                        candidateBroker = broker;
+                    }
+                }
+                if (candidateBroker != null) {
+                    String substring = candidateBroker.substring(candidateBroker.lastIndexOf('/') + 1);
+                    owner = getOrAssignOwnerAsync(bundle, substring).thenApply(Optional::ofNullable);
+                } else {
+                    owner = getOrSelectOwnerAsync(serviceUnit, bundle).thenApply(Optional::ofNullable);
```

Comment:
I don't think we need to publish ownerships of `Heartbeat or SLAMonitor namespace service units` to the ServiceUnitStateChannel because the lookup of `Heartbeat or SLAMonitor namespaces` is static -- lookup should always return the broker parsed from the serviceUnit. (I personally think the client shouldn't trigger lookup in this case but directly connect the broker by parsing the URL).

Here, I think we just need to convert candidateBroker to owner.

Reply:
Yes, we can do this, and I have pushed a commit.